File: backend/src/entities/Meetings.ts
import {
  Column,
  Entity,
  Index,
  OneToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Participants } from './Participants';

@Index('meetingId_UNIQUE', ['meetingId'], { unique: true })
@Entity('meetings', { schema: 'meeting_management' })
export class Meetings {
  @PrimaryGeneratedColumn({ type: 'int', name: 'meetingId', unsigned: true })
  meetingId: number;

  @Column('int', { name: 'roomId' })
  roomId: number;

  @Column('date', { name: 'date' })
  date: string;

  @Column('varchar', { name: 'startTime', length: 10 })
  startTime: string;

  @Column('varchar', { name: 'endTime', length: 10 })
  endTime: string;

  @Column('varchar', { name: 'title', length: 45 })
  title: string;

  @Column('varchar', { name: 'content', length: 200 })
  content: string;

  @Column('varchar', { name: 'repeat', length: 20 })
  repeat: string;

  @OneToMany(() => Participants, (participants) => participants.meeting)
  participants: Participants[];
}

File: frontend/app/lib/model/meetings.dart
import 'package:app/urldata.dart';
import 'package:dio/dio.dart';

import 'meeting.dart';

class Meetings {
  var dio = Dio();
  UrlData urlData = UrlData();
  static Map<String, List<Meeting>> meetings = {};
  Map<String, List<Meeting>> get() => meetings;
  Future<void> getMeeting() async {
    meetings = {};
    await dio.get('${urlData.url}/meetings/all').then((value) {
      for (var i in value.data['data']) {
        if (meetings['${i['date']}'] == null) {
          meetings['${i['date']}'] = [
            Meeting(i['roomId'], i['date'], i['startTime'], i['endTime'],
                i['title'], i['content'], i['repeat'],
                meetingId: i['meetingId'])
          ];
        } else {
          meetings['${i['date']}'] = meetings['${i['date']}']! +
              [
                Meeting(i['roomId'], i['date'], i['startTime'], i['endTime'],
                    i['title'], i['content'], i['repeat'],
                    meetingId: i['meetingId'])
              ];
        }
      }
    });
  }

  Future<void> addMeeting(body) async {
    await dio.post('${urlData.url}/meetings/create', data: body);
  }

  Future<void> updateMeeting(body) async {
    await dio.post('${urlData.url}/meetings/update', data: body);
  }

  Future<bool> deleteMeeting(body) async {
    print(body);
    await dio.post('${urlData.url}/meetings/delete', data: body);
    return true;
  }
}
